### PR TITLE
Refactored Display implementation for Platform enum

### PR DESCRIPTION
**Pull request by sebihutanu**
### Key Changes:
1. **Display Restructuring:**
   - Significantly restructured the Display implementation for the Platform type.
   - Introduced a dedicated file, `disp.rs`, to centralize and streamline platform-specific display functions.

2. **Aesthetic Enhancements:**
   - Revamped the platform display, focusing on enhancing the overall aesthetic.
   - Resulting in a more visually appealing output for improved user experience.
   ##
   To enhance the utility's adaptability, a crucial implementation has been added. The program checks whether it is running in a terminal environment (using https://crates.io/crates/is-terminal) before deciding how to display the platform. If executed in a terminal, the graphical version is presented; otherwise, in scenarios like running in a pipe, the text version is displayed.

### CODE REVIEW

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Please run `cargo fmt` to format your source.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Please fix `clippy`s errors.

File: src/fake.rs [hunk 1]
```diff
@@ -1,75 +1,681 @@
-//! Currently used for WebAssembly unknown (non-web) only
```
Comment (alexandruradovici) [inline src/fake.rs]: Why delete this?

File: src/fake.rs
Comment (alexandruradovici) [inline src/fake.rs:1]: Please explain why do we need this file?

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: src/fake.rs (hunk 1 above)
Comment (sebihutanu) [reply src/fake.rs]: by mistake, will fix

File: src/fake.rs
Comment (sebihutanu) [reply src/fake.rs:1]: I am sorry, I just noticed that I somehow copied the entire code from lib.rs to fake.rs. . I'm not sure how that happened, but I'll fix it immediately.

**Review by AldaronLau — COMMENTED** [review]
Is there a reason that this has to exist within the whoami crate?  I feel like this functionality could be split out into its own crate depending on whoami.

**Comment by AldaronLau** [comment]
@sebihutanu1 I'm going to close this PR, as I have decided it's functionality doesn't belong in the whoami crate.  I recommend creating a separate crate for it, depending on whoami and is-terminal, if you require this functionality.

If you'd like to contribute to this project in another way, feel free to send me an email at <aldaronlau@gmail.com> or comment in #89, and we can figure something out.  Thank you for your interest.

**Pull request closed without merging**